// @prisma/client/primary/index-browser.js

Object.defineProperty(exports, "__esModule", { value: true });

const {
  Decimal,
  objectEnumValues,
  makeStrictEnum,
  Public,
  getRuntime,
  skip
} = require('./runtime/index-browser.js')


const Prisma = {}

exports.Prisma = Prisma
exports.$Enums = {}

/**
 * Prisma Client JS version: 5.22.0
 * Query Engine version: 605197351a3c8bdd595af2d2a9bc3025bca48ea2
 */
Prisma.prismaVersion = {
  client: "5.22.0",
  engine: "605197351a3c8bdd595af2d2a9bc3025bca48ea2"
}

Prisma.PrismaClientKnownRequestError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientKnownRequestError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)};
Prisma.PrismaClientUnknownRequestError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientUnknownRequestError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientRustPanicError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientRustPanicError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientInitializationError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientInitializationError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientValidationError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientValidationError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.NotFoundError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`NotFoundError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.Decimal = Decimal

/**
 * Re-export of sql-template-tag
 */
Prisma.sql = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`sqltag is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.empty = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`empty is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.join = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`join is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.raw = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`raw is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.validator = Public.validator

/**
* Extensions
*/
Prisma.getExtensionContext = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`Extensions.getExtensionContext is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.defineExtension = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`Extensions.defineExtension is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}

/**
 * Shorthand utilities for JSON filtering
 */
Prisma.DbNull = objectEnumValues.instances.DbNull
Prisma.JsonNull = objectEnumValues.instances.JsonNull
Prisma.AnyNull = objectEnumValues.instances.AnyNull

Prisma.NullTypes = {
  DbNull: objectEnumValues.classes.DbNull,
  JsonNull: objectEnumValues.classes.JsonNull,
  AnyNull: objectEnumValues.classes.AnyNull
}



/**
 * Enums
 */

exports.Prisma.TransactionIsolationLevel = makeStrictEnum({
  ReadUncommitted: 'ReadUncommitted',
  ReadCommitted: 'ReadCommitted',
  RepeatableRead: 'RepeatableRead',
  Serializable: 'Serializable'
});

exports.Prisma.PortalAccountScalarFieldEnum = {
  id: 'id',
  username: 'username',
  email: 'email',
  name: 'name',
  passwordHash: 'passwordHash',
  role: 'role',
  headPrivileges: 'headPrivileges',
  companyId: 'companyId',
  staffDesignatedCompanyId: 'staffDesignatedCompanyId',
  staffAssignmentColor: 'staffAssignmentColor',
  customerOrgRole: 'customerOrgRole',
  accountStatus: 'accountStatus',
  profileImage: 'profileImage',
  profileImageZoom: 'profileImageZoom',
  profileImagePosX: 'profileImagePosX',
  profileImagePosY: 'profileImagePosY',
  createdAt: 'createdAt'
};

exports.Prisma.TeamScalarFieldEnum = {
  id: 'id',
  name: 'name',
  description: 'description',
  createdAt: 'createdAt'
};

exports.Prisma.AgentScalarFieldEnum = {
  id: 'id',
  name: 'name',
  email: 'email',
  teamId: 'teamId',
  createdAt: 'createdAt'
};

exports.Prisma.SlaPolicyScalarFieldEnum = {
  id: 'id',
  priority: 'priority',
  firstResponseHours: 'firstResponseHours',
  resolutionHours: 'resolutionHours'
};

exports.Prisma.EscalationTriggerScalarFieldEnum = {
  id: 'id',
  priority: 'priority',
  enabled: 'enabled',
  notifyAdmin: 'notifyAdmin',
  notifyTarget: 'notifyTarget',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.TicketScalarFieldEnum = {
  id: 'id',
  ticketNumber: 'ticketNumber',
  title: 'title',
  description: 'description',
  category: 'category',
  priority: 'priority',
  status: 'status',
  contactName: 'contactName',
  contactEmail: 'contactEmail',
  requestorEmail: 'requestorEmail',
  contactPhone: 'contactPhone',
  teamId: 'teamId',
  assignedAgentId: 'assignedAgentId',
  firstResponseDueAt: 'firstResponseDueAt',
  resolutionDueAt: 'resolutionDueAt',
  firstResponseAt: 'firstResponseAt',
  resolvedAt: 'resolvedAt',
  closedAt: 'closedAt',
  escalationType: 'escalationType',
  escalatedAt: 'escalatedAt',
  reopenCount: 'reopenCount',
  resolutionNotes: 'resolutionNotes',
  intakeScreenshotMeta: 'intakeScreenshotMeta',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.TicketActivityScalarFieldEnum = {
  id: 'id',
  ticketId: 'ticketId',
  actor: 'actor',
  summary: 'summary',
  detail: 'detail',
  createdAt: 'createdAt'
};

exports.Prisma.TicketMessageScalarFieldEnum = {
  id: 'id',
  ticketId: 'ticketId',
  actor: 'actor',
  author: 'author',
  body: 'body',
  createdAt: 'createdAt'
};

exports.Prisma.TicketFeedbackScalarFieldEnum = {
  id: 'id',
  ticketId: 'ticketId',
  csat: 'csat',
  nps: 'nps',
  ces: 'ces',
  comment: 'comment',
  createdAt: 'createdAt'
};

exports.Prisma.KpiMaintenanceScalarFieldEnum = {
  id: 'id',
  title: 'title',
  mainTask: 'mainTask',
  isRecurring: 'isRecurring',
  nonRecurringStartAt: 'nonRecurringStartAt',
  nonRecurringEndAt: 'nonRecurringEndAt',
  frequency: 'frequency',
  subKpis: 'subKpis',
  assignedAgentId: 'assignedAgentId',
  assignedRole: 'assignedRole',
  recurrenceWeekday: 'recurrenceWeekday',
  recurrenceMonthDay: 'recurrenceMonthDay',
  periodCycleStartAt: 'periodCycleStartAt',
  lastFullCompletionAt: 'lastFullCompletionAt',
  periodKey: 'periodKey',
  rolledOverIncomplete: 'rolledOverIncomplete',
  itProjectName: 'itProjectName',
  itProjectPhase: 'itProjectPhase',
  scopedCompanyTeamId: 'scopedCompanyTeamId',
  createdBy: 'createdBy',
  createdByRole: 'createdByRole',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.KpiMaintenancePeriodSnapshotScalarFieldEnum = {
  id: 'id',
  kpiMaintenanceId: 'kpiMaintenanceId',
  periodKey: 'periodKey',
  frequency: 'frequency',
  timeZone: 'timeZone',
  total: 'total',
  done: 'done',
  missing: 'missing',
  percent: 'percent',
  fullyComplete: 'fullyComplete',
  contributorProgress: 'contributorProgress',
  capturedAt: 'capturedAt'
};

exports.Prisma.HelpdeskCsvTicketScalarFieldEnum = {
  id: 'id',
  sheetRowId: 'sheetRowId',
  reportedAt: 'reportedAt',
  resolvedAt: 'resolvedAt',
  statusRaw: 'statusRaw',
  normalizedBucket: 'normalizedBucket',
  userEmail: 'userEmail',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.TaskItemScalarFieldEnum = {
  id: 'id',
  title: 'title',
  description: 'description',
  status: 'status',
  assignedAgentId: 'assignedAgentId',
  priority: 'priority',
  dueAt: 'dueAt',
  createdBy: 'createdBy',
  createdByRole: 'createdByRole',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.TaskActivityScalarFieldEnum = {
  id: 'id',
  taskId: 'taskId',
  author: 'author',
  action: 'action',
  detail: 'detail',
  createdAt: 'createdAt'
};

exports.Prisma.AccountActionRequestScalarFieldEnum = {
  id: 'id',
  portalAccountId: 'portalAccountId',
  requestType: 'requestType',
  reason: 'reason',
  status: 'status',
  reviewedBy: 'reviewedBy',
  reviewedAt: 'reviewedAt',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.SortOrder = {
  asc: 'asc',
  desc: 'desc'
};

exports.Prisma.NullableJsonNullValueInput = {
  DbNull: Prisma.DbNull,
  JsonNull: Prisma.JsonNull
};

exports.Prisma.JsonNullValueInput = {
  JsonNull: Prisma.JsonNull
};

exports.Prisma.QueryMode = {
  default: 'default',
  insensitive: 'insensitive'
};

exports.Prisma.NullsOrder = {
  first: 'first',
  last: 'last'
};

exports.Prisma.JsonNullValueFilter = {
  DbNull: Prisma.DbNull,
  JsonNull: Prisma.JsonNull,
  AnyNull: Prisma.AnyNull
};
exports.TicketPriority = exports.$Enums.TicketPriority = {
  LOW: 'LOW',
  MEDIUM: 'MEDIUM',
  HIGH: 'HIGH',
  URGENT: 'URGENT',
  UNSET: 'UNSET'
};

exports.TicketCategory = exports.$Enums.TicketCategory = {
  IT: 'IT',
  HR: 'HR',
  FINANCE: 'FINANCE',
  OPERATIONS: 'OPERATIONS',
  GENERAL: 'GENERAL'
};

exports.TicketStatus = exports.$Enums.TicketStatus = {
  OPEN: 'OPEN',
  IN_PROGRESS: 'IN_PROGRESS',
  PENDING_INFO: 'PENDING_INFO',
  ESCALATED: 'ESCALATED',
  FOR_CONFIRMATION: 'FOR_CONFIRMATION',
  RESOLVED: 'RESOLVED',
  CLOSED: 'CLOSED'
};

exports.EscalationType = exports.$Enums.EscalationType = {
  FUNCTIONAL: 'FUNCTIONAL',
  HIERARCHICAL: 'HIERARCHICAL'
};

exports.ActivityActor = exports.$Enums.ActivityActor = {
  SYSTEM: 'SYSTEM',
  AGENT: 'AGENT',
  USER: 'USER'
};

exports.KpiFrequency = exports.$Enums.KpiFrequency = {
  DAILY: 'DAILY',
  WEEKLY: 'WEEKLY',
  MONTHLY: 'MONTHLY',
  QUARTERLY: 'QUARTERLY'
};

exports.TaskStatus = exports.$Enums.TaskStatus = {
  CURRENT: 'CURRENT',
  DONE: 'DONE',
  DELAYED: 'DELAYED'
};

exports.Prisma.ModelName = {
  PortalAccount: 'PortalAccount',
  Team: 'Team',
  Agent: 'Agent',
  SlaPolicy: 'SlaPolicy',
  EscalationTrigger: 'EscalationTrigger',
  Ticket: 'Ticket',
  TicketActivity: 'TicketActivity',
  TicketMessage: 'TicketMessage',
  TicketFeedback: 'TicketFeedback',
  KpiMaintenance: 'KpiMaintenance',
  KpiMaintenancePeriodSnapshot: 'KpiMaintenancePeriodSnapshot',
  HelpdeskCsvTicket: 'HelpdeskCsvTicket',
  TaskItem: 'TaskItem',
  TaskActivity: 'TaskActivity',
  AccountActionRequest: 'AccountActionRequest'
};

/**
 * This is a stub Prisma Client that will error at runtime if called.
 */
class PrismaClient {
  constructor() {
    return new Proxy(this, {
      get(target, prop) {
        let message
        const runtime = getRuntime()
        if (runtime.isEdge) {
          message = `PrismaClient is not configured to run in ${runtime.prettyName}. In order to run Prisma Client on edge runtime, either:
- Use Prisma Accelerate: https://pris.ly/d/accelerate
- Use Driver Adapters: https://pris.ly/d/driver-adapters
`;
        } else {
          message = 'PrismaClient is unable to run in this browser environment, or has been bundled for the browser (running in `' + runtime.prettyName + '`).'
        }
        
        message += `
If this is unexpected, please open an issue: https://pris.ly/prisma-prisma-bug-report`

        throw new Error(message)
      }
    })
  }
}

exports.PrismaClient = PrismaClient

Object.assign(exports, Prisma)
